rekor-v2-spec.md

```diff
@@ -0,0 +1,241 @@
+
+A client signing an artifact uploads the entry record, formatted as either a
+[HashedRekordRequestV002](https://github.com/sigstore/rekor-tiles/blob/f3cba09c2f92f1d2d5a7ca2b5694b66941a82d41/api/proto/rekor/v2/hashedrekord.proto#L31C9-L31C32)
+or
+[DSSERequestV002](https://github.com/sigstore/rekor-tiles/blob/f3cba09c2f92f1d2d5a7ca2b5694b66941a82d41/api/proto/rekor/v2/dsse.proto#L32C9-L32C24) to the
+`/api/v2/log/entries` (HTTP) or `dev.sigstore.rekor.v2.Rekor.CreateEntry` (gRPC)
+service. The client receives, as a response, a
+[TransparencyLogEntry](https://github.com/sigstore/protobuf-specs/blob/4df5baadcdb582a70c2bc032e042c0a218eb3841/protos/sigstore_rekor.proto#L94),
+which contains the latest checkpoint and an inclusion proof for the entry.
+Because entries are batched before being integrated, and because checkpoints are
+published only at regular intervals, the client may have to wait for a few
+seconds before receiving a response, depending on the configuration of the
+service.
+
+### 5.2 Artifact Verifiers
+
+A client verifying an artifact MUST NOT perform an online lookup of the entry.
+The signing client must provide the entry's checkpoint, inclusion proof,
+verification key and digest in the verification material it provides for the
+verifying client. The verifying client uses the material it is provided with and
+the public key of the log to verify the entry's inclusion in the log.
+
+### 5.3 Monitors
+
+A monitor is a client that monitors the contents of a transparency log to
+protect a given user against entry key compromise. If a key or identity is found
+to have been used in a signing event, it is done so in the open, and a monitor
+detects it and reports it to the user to whom the key or identity belongs. The
+monitor uses the tile entries API at `/api/v2/tile/entries` to tail the log for
+its contents. A monitor could also periodically verify the consistency of the
+log.
+
+### 5.4 Witnesses
+
+Synchronous witnesses are responsible for verifying the consistency of a log.
+Their co-signatures of a checkpoint are returned to the log for distribution to
+signing clients to use in their verification material bundle. Witnessing is
+described in the [Transparency Log Witness
+Protocol](https://github.com/C2SP/C2SP/blob/main/tlog-witness.md).
+
+Verifying the log's consistency is essential for proving that a log remains
+append-only and has not been tampered with. Using a diverse network of
+corroborating witnesses helps to mitigate split-view attacks, where a malicious
+log could present alternate views of a tree to different parties.
+
+## Privacy
+
+Rekor v2 has most of the same privacy considerations as [Rekor
+v1](./rekor-spec.md#6-privacy), such as that sensitive content may be recorded
+and publicly viewable in the log and parties uploading entries should be aware
+of this potential exposure. Rekor v2 has a privacy advantage over Rekor v1 in
+that online requests for inclusion proofs are not supported, so there is no
```

Comment:
Same here, online requests are supported, just clients have to compute which tiles are needed to build an inclusion proof. It's that there's no longer an API for requesting an inclusion proof.

Reply:
:+1: